cli/utils.go

```diff
@@ -0,0 +1,209 @@
+package cli
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/spf13/viper"
+
+	"github.com/prometheus/alertmanager/types"
+)
+
+const (
+	RESOLVE_EXACT = iota
+	RESOLVE_FUZZY = iota
+)
+
+func GetAlertmanagerURL() (*url.URL, error) {
+	u, err := url.ParseRequestURI(viper.GetString("alertmanager.url"))
+	if err != nil {
+		return nil, errors.New("Invalid alertmanager url")
+	}
+	return u, nil
+}
+
+// Parse a list of labels (cli arguments)
+func parseMatchers(labels []string, resolve int) (types.Matchers, error) {
```

Comment:
Filtering silences is available in that pr, in https://github.com/prometheus/alertmanager/pull/633/commits/0954959e821f76ef37b925a4f819d8ceff79233d

I'm going to try to get that PR pushed through, then it should just be a few easy changes here and we can get this merged.